include/polycap-profile.h

```diff
@@ -106,6 +106,48 @@ polycap_profile* polycap_profile_new_from_file(
  */
 int polycap_profile_validate(polycap_profile *profile, int64_t n_cap, polycap_error **error);
 
+
+/* Allows the user to set a profile shape given the appropriate shape parameter arrays
+ *
+ * \param profile a polycap_profile
+ * \param nid amount of elements in the provided array. It is suggested to have at least 1000 elements.
+ * \param ext external polycapillary shape profile radii [cm].
+ * \param cap central capillary shape profile radii [cm].
+ * \param z Z-coordinates running from 0 to polycapillary length [cm].
+ */ 
+POLYCAP_EXTERN
+void polycap_profile_set_profile(polycap_profile *profile, int nid, double *ext, double *cap, double *z, polycap_error **error);
+
+/** Retrieve exit electric field vector from a polycap_photon
+ * 
+ * \param profile a polycap_profile
+ * \param nid amount of elements in the array.
+ * \param ext external polycapillary shape profile radii [cm].
+ * \returns true on success, otherwise fail.
+ */
+POLYCAP_EXTERN
+bool polycap_profile_get_ext(polycap_profile *profile, size_t *nid, double **ext);
```

Comment:
Misschien best ook `polycap_error **error` toevoegen aan het einde van deze methods om makkelijker met exceptions te kunnen werken in python?